akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -540,27 +562,51 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
     if (activeInterpreters.isEmpty) context.stop(self)
   }
 
+  private def shortCircuitBatch(): Unit = {
+    @tailrec
+    def shortCircuitBatchUtil(): Unit =
+      if (!shortCircuitBuffer.isEmpty) {
+        val input = shortCircuitBuffer.dequeue()
+        processEvent(input)
+        shortCircuitBatchUtil()
```

Comment:
okay, that happens to make it bounded, but it limits it in a non-intuitive way (and not to the shortCircuitLimit, rather to twice as much)